ktor-utils/common/src/io/ktor/util/Attributes.kt

```diff
@@ -4,24 +4,27 @@
 
 package io.ktor.util
 
+import kotlin.jvm.*
 import kotlin.reflect.*
 
 /**
  * Specifies a key for an attribute in [Attributes]
  * @param T is a type of the value stored in the attribute
  * @param name is a name of the attribute for diagnostic purposes. Can't be blank
  */
+@JvmSynthetic
 public inline fun <reified T : Any> AttributeKey(name: String): AttributeKey<T> =
-    AttributeKey(name, T::class.toString())
+    AttributeKey(name, typeOf<T>())
```

Comment:
please check if it works on wasm and js

Reply:
The tests pass ✅ 

One of the few platforms that actually pass... 😞